sdk/batch/Azure.ResourceManager.Batch/tests/Helpers/ResourceDataHelper.cs

```diff
@@ -0,0 +1,182 @@
+        };
+            return data;
+        }
+        public static void AssertCertificate(BatchAccountCertificateData certificateData1, BatchAccountCertificateData certificateData2)
+        {
+            AssertResourceData(certificateData1, certificateData2);
+            Assert.AreEqual(certificateData1.ETag, certificateData2.ETag);
+        }
+        #endregion
+
+        #region Account
+        public static BatchAccountCreateOrUpdateContent GetBatchAccountData()
+        {
+            var data = new BatchAccountCreateOrUpdateContent(AzureLocation.WestUS)
+            {
+                AutoStorage = new BatchAccountAutoStorageBaseConfiguration(new ResourceIdentifier("/subscriptions/db1ab6f0-4769-4b27-930e-01e2ef9c123c/resourceGroups/AutoRestResources2/providers/Microsoft.Storage/storageAccounts/20220725datafactory"))
+            };
+            return data;
+        }
+        public static void AssertAccount(BatchAccountData account1, BatchAccountData account2)
+        {
+            //AssertResourceData(account1, account2);
+            Assert.AreEqual(account1.Tags, account2.Tags);
```

Comment:
Since Tags is a collection, we cannot assert they have the same content like this.
Please try use methods from CollectionAssert

Reply:
Also we should add more assertions

Reply:
fixed